specification/api-context.md

```diff
@@ -0,0 +1,90 @@
+# Context
```

Comment:
I feel this could be called an SDK API. These APIs will not be used directly by users, they're used to implement the cross-cutting concerns. I fear that mention of "key" here will confuse people, since it sounds like we're accessing parts of the correlation context by key w/o more clarity.

Reasons to call this an SDK API: the actual uses of this API are protected by the packages that implement the cross-cutting concerns. so the SDK has helpers to set and get the current (Span, Upstream context, Correlation Context) -- it will call these APIs on your behalf. The SDK will be written to call those helpers, not these APIs. So the only people who need to understand these APIs are SDK authors who will write helpers. There's the notion that providing helpers isolates the context keys, so that we don't have to make particularly strong statements about what kind of value they are. Users shouldn't even know this exists--we don't need to worry about users poking around in the context for fields they shouldn't access.